src/common/delivery/goals/SdmGoalImplementationMapper.ts

```diff
@@ -102,7 +112,15 @@ export class SdmGoalImplementationMapper {
         return undefined;
     }
 
-    public findFullfillmentCallbackForGoal(goal: SdmGoal): GoalFullfillmentCallback[] {
-        return this.callbacks.filter(c => c.goalTest(goal));
+    public findFullfillmentCallbackForGoal(g: SdmGoal): GoalFullfillmentCallback[] {
+        return this.callbacks.filter(c => c.goalTest(g));
+    }
+
+    public getIsolatedGoalLauncher(): IsolatedGoalLauncher {
```

Comment:
Should this be a constructor argument to make it easier to test and make parameterization possible?

Reply:
changed. for now we only support the kubernetes show this on is used from softwareDeliveryMachine.